### ISSUE TITLE
Remove --recompile-scripts

### ISSUE THREAD

**Issue by oehme** [gradle/gradle#6306]

**Comment by marcphilipp** [comment]
@gradle/kotlin-dsl As part of removing the option I'm going to deprecate `StartParameter.isRecompileScript` which is used by the Kotlin DSL in https://github.com/gradle/kotlin-dsl/blob/master/subprojects/provider/src/main/kotlin/org/gradle/kotlin/dsl/cache/BuildServices.kt#L41. Shall I remove the option from there, too?

**Comment by Vampire** [comment]
Why was this removed?
Is there a replacement?
Besides that is has a bug that it only works for the file-based cache, not for the in-memory cache (guessed from observations), it was the only way besides manually deleting cache files and restarting the daemon to overcome a Gradle bug (#7992).

**Comment by marcphilipp** [comment]
It was deprecated for quite some time and thus removed for 5.0. Thanks for reporting #7992, which will be fixed in 5.2.

**Comment by Vampire** [comment]
Yes, I know your deprecation policy. Let me rephrase my question. Why was this deprecated? Bugs can happen and it was quite handy to have an easy switch that makes the scripts recompile if such a bug was triggered. I also observe another bug that happens from time to time at colleagues or at my box where Gradle thinks a task is up to date which it is not. When this happens a clean helps for the symptom, but after executing the task once it starts to be falsely up to date again. Only deleting the cache helps to restore proper operation. But there I have even less information about why or when this happens which is why I didn't report it yet. This has of course nothing to do with the recompile scripts parameter, I just mention it to show that sometimes strange bugs simply happen and that it is handy to have an easy emergency option. :-)

**Comment by marcphilipp** [comment]
Sounds like it was deprecated because it didn't work correctly when used with the Daemon: #1425

**Comment by oehme** [comment]
And thanks to the removal you reported the bug (and we probably fixed it with my latest change) instead of working around it. I think that's a better outcome for everyone.

**Comment by oehme** [comment]
Opening it can't hurt, with as much info as you have. It'll be hard to do anything about it without a reproducible case, but maybe the symptoms alone are enough, like with this one.

**Comment by Vampire** [comment]
Ok, there you have it: #8035 :-)
